src/s3/args.rs

```diff
@@ -1048,12 +1048,10 @@ impl<'a> ListObjectsV2Args<'a> {
     ///
     /// ```
     /// use minio::s3::args::*;
-    /// let args = ListObjectsV2Args::new("my-bucket").unwrap();
+    /// let args = ListObjectsV2Args::new("my-bucket");
     /// ```
-    pub fn new(bucket_name: &'a str) -> Result<ListObjectsV2Args<'a>, Error> {
-        check_bucket_name(bucket_name, true)?;
-
-        Ok(ListObjectsV2Args {
+    pub fn new(bucket_name: &'a str) -> ListObjectsV2Args<'a> {
```

Comment:
I am thinking we do this bucket validity check commonly in `client.execute` so that all these constructors do not have to return `Result<_, Error>`

Reply:
That's not a good idea. We should stop on invalid values right away to avoid further processing. There are many other parameters in different APIs to validate arguments in `new()`. Having different behavior of different APIs will look odd.